pkg/bot/teams.go

```diff
@@ -46,61 +47,59 @@ const (
 
 var _ Bot = &Teams{}
 
+type conversation struct {
+	ref    schema.ConversationReference
+	notify bool
+}
+
 // Teams listens for user's message, execute commands and sends back the response.
 type Teams struct {
 	log             logrus.FieldLogger
 	executorFactory ExecutorFactory
 	reporter        AnalyticsReporter
-	notifyMutex     sync.RWMutex
-	notify          bool
-
-	BotName          string
-	AppID            string
-	AppPassword      string
-	MessagePath      string
-	Port             string
-	AllowKubectl     bool
-	RestrictAccess   bool
-	ClusterName      string
-	Notification     config.Notification
-	Adapter          core.Adapter
-	DefaultNamespace string
-
-	conversationRefMutex sync.RWMutex
-	conversationRef      *schema.ConversationReference
+	// TODO: Be consistent with other communicators when Teams supports multiple channels
```

Comment:
I thought that:
1. User sets proper bindings in values YAML with expected channelIDs
2. When we get the "conversation" activity we check if we have a given ID on our list
    2.1 If yes, use a proper binding list
    2.2 If not, set `isAuthChannel` to `false` run executor 

Is this sth that can be implemented?

I know that it's different from Slack approach where user don't need to start the conversation, however if already started we should get sth that identifies a given channel (id/name)

Reply:
Unfortunately it's hard to get ID from Teams UI (link for channel contains UUIDs, but the Activity and channel IDs are numeric ones). As discussed, it needs to stay as it is 😞